web/src/screens/filters/details.tsx

```diff
@@ -144,37 +145,49 @@ export default function FilterDetails() {
   const navigate = useNavigate();
   const { filterId } = useParams<{ filterId: string }>();
 
-  const { isLoading, data: filter } = useQuery(
-    ["filters", filterId],
-    () => APIClient.filters.getByID(parseInt(filterId ?? "0")),
-    {
-      retry: false,
-      refetchOnWindowFocus: false,
-      onError: () => navigate("./")
+  if (filterId === "0" || undefined) {
```

Comment:
filterId should never be undefined. Since we match for "/*" after it.

Personally, I wouldn't hard-code navigate-back checks like this, because they will always miss on upper bound (if id >= max filter id), so I think it would be best if we made the HttpClient function respect relative redirects in those cases. And we also don't notify the user in this case, which is kinda eh (even though this case is extremely improbable).

Reply:
It's ugly indeed, but with the planned switch, or at least exploration of Tanstack router then that will be an issue of the past.

react-router have done some changes as well into similar direction but it's getting more and more tied in with Remix.